src/network/serverpackethandler.cpp

```diff
@@ -1080,19 +1079,20 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 		Make sure the player is allowed to do it
 	*/
 	if (!checkPriv(player->getName(), "interact")) {
-		actionstream<<player->getName()<<" attempted to interact with "
-				<<pointed.dump()<<" without 'interact' privilege"
-				<<std::endl;
-		// Re-send block to revert change on client-side
-		RemoteClient *client = getClient(pkt->getPeerId());
-		// Digging completed -> under
-		if (action == 2) {
-			v3s16 blockpos = getNodeBlockPos(floatToInt(pointed_pos_under, BS));
-			client->SetBlockNotSent(blockpos);
-		}
-		// Placement -> above
-		else if (action == 3) {
-			v3s16 blockpos = getNodeBlockPos(floatToInt(pointed_pos_above, BS));
+		actionstream << player->getName() << " attempted to interact with "
+			<< pointed.dump() << " without 'interact' privilege" << std::endl;
+		// This may only happen while revoking interact before the client
+		// receives its change of privileges
+		// Re-send the whole block to undo placement or dig prediction
+		if (action == 2 || action == 3) {
+			RemoteClient *client = getClient(peer_id);
+			v3s16 blockpos;
```

Comment:
v3s16 blockpos = getNodeBlockPos(action == 2 ? p_under : p_above);

Reply:
This code actually does not work right: For placement under and above need to be sent.
For a long time interacting without interact hasn't been possible (except the rare edge case while revoking interact), I guess mapblock sending can be removed here.